component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -683,4 +700,64 @@ private boolean validateJTIInCache(String jti, SignedJWT signedJWT, JWTCacheEntr
+     * Validate whether the request signing algorithm is configured for the application.
+     *
+     * @param signedJWT     The signed JWT.
+     * @param clientId      Client ID of the application.
+     * @return whether the request signing algorithm is configured for the application.
+     * @throws OAuthClientAuthnException OAuth Client Authentication Exception.
+     */
+    private boolean isValidSignatureAlgorithm(SignedJWT signedJWT, String clientId) throws OAuthClientAuthnException {
+
+        //   Obtain the signing algorithm used to sign the JWT in the request.
+        String requestSigningAlgorithm = signedJWT.getHeader().getAlgorithm().getName();
+        //   Obtain the signing algorithm configured for the application.
+        List<String> configuredSigningAlgorithms = getConfiguredSigningAlgorithm(clientId);
+        //   Mandating FAPI specified JWT signing algorithms.
+        List<String> fapiAllowedSigningAlgorithms = IdentityUtil.getPropertyAsList(FAPI_SIGNATURE_ALG_CONFIGURATION);
+        if (!fapiAllowedSigningAlgorithms.contains(requestSigningAlgorithm)) {
+            if (log.isDebugEnabled()) {
+                log.debug("FAPI unsupported signing algorithm " + requestSigningAlgorithm +
+                        " is used to sign the JWT.");
+            }
+            return false;
+        }
+        //  Validate whether the JWT signing algorithm is configured for the application.
+        if (configuredSigningAlgorithms.contains(requestSigningAlgorithm)) {
+            return true;
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("JWT signed algorithm does not match with the configured algorithms: " +
+                        configuredSigningAlgorithms);
+            }
+            return false;
+        }
+    }
+
+    /**
+     * Obtain the request signing algorithms configured for the application.
+     *
+     * @param clientId   Client ID of the application.
+     * @throws OAuthClientAuthnException OAuth Client Authentication Exception.
+     */
+    private List<String> getConfiguredSigningAlgorithm(String clientId) throws OAuthClientAuthnException {
+
+        try {
+            ServiceProvider serviceProvider = OAuth2Util.getServiceProvider(clientId);
+            ServiceProviderProperty[] serviceProviderProperties = serviceProvider.getSpProperties();
+            for (ServiceProviderProperty serviceProviderProperty : serviceProviderProperties) {
+                if (Constants.TOKEN_ENDPOINT_AUTH_SIGNING_ALG.equals(serviceProviderProperty.getName())) {
+                    return Arrays.asList(serviceProviderProperty.getValue());
+                }
+            }
+        } catch (IdentityOAuth2Exception e) {
+            throw new OAuthClientAuthnException("Error occurred while retrieving the service provider.",
+                    OAuth2ErrorCodes.INVALID_REQUEST, e);
+        }
+
+        return IdentityUtil.getPropertyAsList(FAPI_SIGNATURE_ALG_CONFIGURATION);
```

Comment:
With https://github.com/wso2-extensions/identity-oauth-addons/pull/108/files#r1348854411 lets return a null from here and at the check level do a null check and if no sig alg is configured for the app skip the validation. As we can think as no value is configured that any value that can be processed by the server is accepted

Reply:
Fixed in f62c438a62cda25d10b408fb266521c3cd30aaec and 393f070cb3d7c0feb53e15516f6e562bddb36dee